src/java/com/zimbra/cs/taglib/bean/BeanUtils.java

```diff
@@ -1971,6 +1976,32 @@ public static void checkWebClientEnabled(ZMailbox mbox) throws ServiceException
             throw AccountServiceException.WEB_CLIENT_ACCESS_NOT_ALLOWED(mbox.getName());
         }
     }
+
+    public static boolean isTwoFactorAuthRequired(PageContext pc) throws Exception {
+        ZAuthToken zAuthToken;
+        try {
+            zAuthToken = ZJspSession.getAuthToken(pc);
+        } catch (Exception e) {
+            return false;
+        }
+        return isTwoFactorAuthRequired(zAuthToken);
+    }
+
+    public static boolean isTwoFactorAuthRequired(ZAuthToken zAuthToken) {
+        try {
+            String authtoken = zAuthToken.getValue();
+            String[] tokenParts = authtoken.split("_");
+            String target = tokenParts[2];
+            Map<?, ?> decodedTokenMap = TokenUtil.getAttrs(target);
+            String usage = (String) decodedTokenMap.get(AuthTokenProperties.C_USAGE);
+            if (Usage.TWO_FACTOR_AUTH.getCode().equals(usage)) {
+                return true;
+            }
+        } catch (Exception e) {
+            // no authtoken, parse error etc.
+        }
```

Comment:
More clarity in why nothing is happening in the catch() block would be preferred. I initially read the comment as more of a TODO.